pkg/featuregates/featuregates.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package featuregates
+
+import (
+	"errors"
+
+	"github.com/azure/kaito/pkg/utils/consts"
+	cliflag "k8s.io/component-base/cli/flag"
+)
+
+var (
+	// FeatureGates is a map that holds	the feature gates and their values for Kaito.
+	FeatureGates = map[string]bool{
+		consts.FeatureFlagKarpenter: false,
+	}
+)
+
+// ParseAndValidateFeatureGates parses the feature gates flag and sets the environment variables for each feature.
+func ParseAndValidateFeatureGates(featureGates string) error {
+	gateMap := map[string]bool{}
+
+	if err := cliflag.NewMapStringBool(&gateMap).Set(featureGates); err != nil {
+		return err
+	}
+	if len(gateMap) == 0 {
+		// no feature gates set
+		return nil
+	}
+	if val, ok := gateMap[consts.FeatureFlagKarpenter]; ok {
```

Comment:
You'd do sth like this:

for k, v := range gateMap {
       if _, ok := FeatureGates[k]; !ok {
              return errors.New("invalid feature gate")
       }
       FeatureGates[k]=v
}

Reply:
Addressed but with a return of all invalid features. Added more UT.